openlibrary/catalog/add_book/load_book.py

```diff
@@ -220,10 +220,9 @@ def get_redirected_authors(authors: list["Author"]):
         seen.add(key)
         assert a.type.key == '/type/author'
         # Both records are dateless: assume a potential match
-        if not has_dates(author) and not has_dates(a):
-            match.append(a)
-        # Match if both records have at least one date, and the dates are compatible:
-        elif has_dates(author) and has_dates(a) and author_dates_match(author, a):
+        if (not has_dates(author) and not has_dates(a)) or (
+            has_dates(author) and has_dates(a) and author_dates_match(author, a)
+        ):
```

Comment:
I really don't want this auto commit that deletes half the comments and messes up what I was trying to calrify.
The logic is confusing enough as it is, perhaps I should wrap the two branches in unifying methods, but having all the logic in one place was already pretty readable.

I don't like fighting with pre-commit ci bot, and that's a regular feature of any PR I make to OL.

This one is particularly bad, and I don't know how to undo it or prevent it. @cdrini do you know how to fix this?

It's doubly bad because AFAICT it accepted this code when I first submitted it in 527a1b3fdb80373b69599296538c8df69b7200e9 and then rejected it after a change to a different file in da438aa0a59294ad9afb2536706da6399882c543